### PR TITLE
Feature/4.3 unified view api update

### PR DESCRIPTION
**Pull request by LukeVincentCivica**
Add new endpoints for unified view work.

### CODE REVIEW

File: src/db/migrations/v1.9.0-add-unified-view-collected.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"use strict";
```
Comment (off-beat) [inline src/db/migrations/v1.9.0-add-unified-view-collected.js]: this migration isn't needed any more

File: src/db/models/Registration.js [hunk 2]
```diff
@@ -19,6 +19,13 @@ module.exports = (sequelize, DataTypes) => {
       },
       collected_at: {
         type: DataTypes.STRING
+      },
```
Comment (off-beat) [inline src/db/models/Registration.js]: this isn't needed any more - can cancel PR in reg-service too

File: src/api/registrations/registrations.controller.test.js [hunk 3]
```diff
@@ -1,34 +1,41 @@
 jest.mock("../../connectors/registrationDb/registrationDb.connector", () => ({
-  getAllRegistrations: jest.fn(),
+  getAllRegistrationsByCouncil: jest.fn(),
+  getUnifiedRegistrations: jest.fn(),
   getSingleRegistration: jest.fn(),
-  updateRegistrationCollected: jest.fn(),
-  registrationDbDouble: jest.fn()
+  updateRegistrationCollectedByCouncil: jest.fn(),
+  updateAllRegistrationsCollectedByUnified: jest.fn(),
```
Comment (off-beat) [inline src/api/registrations/registrations.controller.test.js]: not needed - few instances below

File: src/api/registrations/registrations.controller.js [hunk 4]
```diff
@@ -75,6 +76,39 @@ const getRegistration = async options => {
...
+
+  if (validationResult === true) {
+    if (options.double_mode) {
+      return registrationDbDouble(options.double_mode);
+    }
+
+    const registrations = await getUnifiedRegistrations(
+      options.before,
+      options.after,
+      ["establishment", "metadata"]
```
Comment (off-beat) [inline src/api/registrations/registrations.controller.js]: Shouldn't we be returning all fields here - either pass them all here in or just always return all from getUnifiedRegistrations

Comment (LukeVincentCivica) [reply src/api/registrations/registrations.controller.js]: This does pass in all of the fields, things like operator and premise are part of establishment. I've try to keep consistent with the previous functionality for getting the registrations for the council where the fields are passed in from the controller level. But as the council can specify the fields required it get's passed in as an option, where for unified it's hard coded as unified always wants everything

File: src/connectors/registrationDb/registrationDb.double.js [hunk 5]
```diff
@@ -86,13 +86,22 @@ const doubleResponse = [
...
+];
+
 const registrationDbDouble = double_mode => {
   if (double_mode === "success") {
     return doubleResponse;
   } else if (double_mode === "fail") {
     throw new Error("Registration DB double, double mode is set to 'fail'");
   } else if (double_mode === "update") {
     return updateResponse;
+  } else if (double_mode === "updateUnified") {
```
Comment (off-beat) [inline src/connectors/registrationDb/registrationDb.double.js]: no longer needed

File: tests/component/getUnifiedRegistrations.component.test.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
+require("dotenv").config();
+const request = require("request-promise-native");
+
+const baseUrl = process.env.COMPONENT_TEST_BASE_URL || "http://localhost:4001";
+const url = `${baseUrl}/api/registrations/unified`;
+
+describe("GET to /api/registrations/unified", () => {
```
Comment (off-beat) [inline tests/component/getUnifiedRegistrations.component.test.js]: maybe add a test that sets dates in the future to see no records are returned and same with dates in past. Best we can do now without seeding with specific dates.

**Pull request merged**